ostd/src/task/processor.rs

```diff
@@ -99,10 +109,14 @@ pub(super) fn switch_to_task(next_task: Arc<Task>) {
     // next task. Not dropping is just fine because the only consequence is that we delay the drop
     // to the next task switching.
 
+    // Set the previous running task to background.
+    // See also `kernel_task_entry`.
+    if let Some(current) = Task::current() {
```

Comment:
I believe you meant to clear the foreground flag of the _previous_ task, not the _current_ task. To do this, you need to access the `PREVIOUS_TASK_PTR` CPU-local variable.